File: rust/tests/gilded_rose_tests.rs
use gilded_rose::{gilded_rose::GildedRose, item::Item};

mod normal_item {
    use super::*;

    #[test]
    fn before_sell_date() {
        let mut items = vec![Item::new("randomstring", 5, 10)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 4);
        assert_eq!(items[0].quality, 9);
    }

    #[test]
    fn with_min_quality() {
        let mut items = vec![Item::new("randomstring", 5, 0)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 4);
        assert_eq!(items[0].quality, 0);
    }

    #[test]
    fn on_sell_date() {
        let mut items = vec![Item::new("randomstring", 0, 10)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, -1);
        assert_eq!(items[0].quality, 8);
    }

    #[test]
    fn on_sell_date_with_min_quality() {
        let mut items = vec![Item::new("randomstring", 0, 0)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, -1);
        assert_eq!(items[0].quality, 0);
    }

    #[test]
    fn on_sell_date_near_min_quality() {
        let mut items = vec![Item::new("randomstring", 0, 1)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, -1);
        assert_eq!(items[0].quality, 0);
    }

    #[test]
    fn after_sell_date() {
        let mut items = vec![Item::new("randomstring", -10, 10)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, -11);
        assert_eq!(items[0].quality, 8);
    }

    #[test]
    fn after_sell_date_with_min_quality() {
        let mut items = vec![Item::new("randomstring", -10, 0)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, -11);
        assert_eq!(items[0].quality, 0);
    }

    #[test]
    fn after_sell_date_near_min_quality() {
        let mut items = vec![Item::new("randomstring", -10, 1)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, -11);
        assert_eq!(items[0].quality, 0);
    }
}

mod aged_cheddar {
    use super::*;

    #[test]
    fn before_sell_date() {
        let mut items = vec![Item::new("Aged Cheddar", 5, 10)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 4);
        assert_eq!(items[0].quality, 11);
    }

    #[test]
    fn with_max_quality() {
        let mut items = vec![Item::new("Aged Cheddar", 5, 50)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 4);
        assert_eq!(items[0].quality, 50);
    }

    #[test]
    fn on_sell_date() {
        let mut items = vec![Item::new("Aged Cheddar", 0, 10)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, -1);
        assert_eq!(items[0].quality, 12);
    }

    #[test]
    fn on_sell_date_with_max_quality() {
        let mut items = vec![Item::new("Aged Cheddar", 0, 50)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, -1);
        assert_eq!(items[0].quality, 50);
    }

    #[test]
    fn on_sell_date_near_max_quality() {
        let mut items = vec![Item::new("Aged Cheddar", 0, 49)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, -1);
        assert_eq!(items[0].quality, 50);
    }

    #[test]
    fn after_sell_date() {
        let mut items = vec![Item::new("Aged Cheddar", -10, 10)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, -11);
        assert_eq!(items[0].quality, 12);
    }

    #[test]
    fn after_sell_date_with_max_quality() {
        let mut items = vec![Item::new("Aged Cheddar", -10, 50)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, -11);
        assert_eq!(items[0].quality, 50);
    }

    #[test]
    fn after_sell_date_near_max_quality() {
        let mut items = vec![Item::new("Aged Cheddar", -10, 49)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, -11);
        assert_eq!(items[0].quality, 50);
    }
}

mod hammer {
    use super::*;

    #[test]
    fn before_sell_date() {
        let mut items = vec![Item::new("Hammer", 5, 40)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 5);
        assert_eq!(items[0].quality, 40);
    }

    #[test]
    fn on_sell_date() {
        let mut items = vec![Item::new("Hammer", 0, 40)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 0);
        assert_eq!(items[0].quality, 40);
    }

    #[test]
    fn after_sell_date() {
        let mut items = vec![Item::new("Hammer", -10, 40)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, -10);
        assert_eq!(items[0].quality, 40);
    }
}

mod concert_tickets {
    use super::*;
    #[test]
    fn long_before_sell_date() {
        let mut items = vec![Item::new("Concert Tickets", 11, 10)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 10);
        assert_eq!(items[0].quality, 11);
    }

    #[test]
    fn long_before_sell_date_at_max_quality() {
        let mut items = vec![Item::new("Concert Tickets", 11, 50)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 10);
        assert_eq!(items[0].quality, 50);
    }

    #[test]
    fn medium_close_to_sell_date_upper_bound() {
        let mut items = vec![Item::new("Concert Tickets", 10, 10)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 9);
        assert_eq!(items[0].quality, 12);
    }

    #[test]
    fn medium_close_to_sell_date_upper_bound_at_max_quality() {
        let mut items = vec![Item::new("Concert Tickets", 10, 50)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 9);
        assert_eq!(items[0].quality, 50);
    }

    #[test]
    fn medium_close_to_sell_date_upper_bound_near_max_quality() {
        let mut items = vec![Item::new("Concert Tickets", 10, 49)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 9);
        assert_eq!(items[0].quality, 50);
    }

    #[test]
    fn medium_close_to_sell_date_lower_bound() {
        let mut items = vec![Item::new("Concert Tickets", 6, 10)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 5);
        assert_eq!(items[0].quality, 12);
    }

    #[test]
    fn medium_close_to_sell_date_lower_bound_at_max_quality() {
        let mut items = vec![Item::new("Concert Tickets", 6, 50)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 5);
        assert_eq!(items[0].quality, 50);
    }

    #[test]
    fn medium_close_to_sell_date_lower_bound_near_max_quality() {
        let mut items = vec![Item::new("Concert Tickets", 6, 49)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 5);
        assert_eq!(items[0].quality, 50);
    }

    #[test]
    fn very_close_to_sell_date_upper_bound() {
        let mut items = vec![Item::new("Concert Tickets", 5, 10)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 4);
        assert_eq!(items[0].quality, 13);
    }

    #[test]
    fn very_close_to_sell_date_upper_bound_at_max_quality() {
        let mut items = vec![Item::new("Concert Tickets", 5, 50)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 4);
        assert_eq!(items[0].quality, 50);
    }

    #[test]
    fn very_close_to_sell_date_upper_bound_near_max_quality() {
        let mut items = vec![Item::new("Concert Tickets", 5, 48)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 4);
        assert_eq!(items[0].quality, 50);
    }

    #[test]
    fn very_close_to_sell_date_lower_bound() {
        let mut items = vec![Item::new("Concert Tickets", 1, 10)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 0);
        assert_eq!(items[0].quality, 13);
    }

    #[test]
    fn very_close_to_sell_date_lower_bound_at_max_quality() {
        let mut items = vec![Item::new("Concert Tickets", 1, 50)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 0);
        assert_eq!(items[0].quality, 50);
    }

    #[test]
    fn very_close_to_sell_date_lower_bound_near_max_quality() {
        let mut items = vec![Item::new("Concert Tickets", 1, 48)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, 0);
        assert_eq!(items[0].quality, 50);
    }

    #[test]
    fn on_sell_date() {
        let mut items = vec![Item::new("Concert Tickets", 0, 10)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, -1);
        assert_eq!(items[0].quality, 0);
    }

    #[test]
    fn after_sell_date() {
        let mut items = vec![Item::new("Concert Tickets", -10, 10)];
        let gilded_rose = GildedRose::new();

        gilded_rose.process_end_of_day(&mut items);

        assert_eq!(items[0].days_remaining, -11);
        assert_eq!(items[0].quality, 0);
    }
}

File: dotnetcore/GildedRose.Tests/GildedRoseTests.cs
using System.Collections.Generic;
using Xunit;

namespace GildedRose.Tests;

public class GildedRoseTests
{
    public class NormalItemTests
    {
        [Fact]
        public void BeforeSellDate()
        {
            IList<Item> items = new List<Item> { new Item { Name = "randomstring", DaysRemaining = 5, Quality = 10 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(4, items[0].DaysRemaining);
            Assert.Equal(9, items[0].Quality);
        }

        [Fact]
        public void WithMinQuality()
        {
            IList<Item> items = new List<Item> { new Item { Name = "randomstring", DaysRemaining = 5, Quality = 0 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(4, items[0].DaysRemaining);
            Assert.Equal(0, items[0].Quality);
        }

        [Fact]
        public void OnSellDate()
        {
            IList<Item> items = new List<Item> { new Item { Name = "randomstring", DaysRemaining = 0, Quality = 10 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(-1, items[0].DaysRemaining);
            Assert.Equal(8, items[0].Quality);
        }

        [Fact]
        public void OnSellDateWithMinQuality()
        {
            IList<Item> items = new List<Item> { new Item { Name = "randomstring", DaysRemaining = 0, Quality = 0 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(-1, items[0].DaysRemaining);
            Assert.Equal(0, items[0].Quality);
        }

        [Fact]
        public void OnSellDateNearMinQuality()
        {
            IList<Item> items = new List<Item> { new Item { Name = "randomstring", DaysRemaining = 0, Quality = 1 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(-1, items[0].DaysRemaining);
            Assert.Equal(0, items[0].Quality);
        }

        [Fact]
        public void AfterSellDate()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "randomstring", DaysRemaining = -10, Quality = 10 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(-11, items[0].DaysRemaining);
            Assert.Equal(8, items[0].Quality);
        }

        [Fact]
        public void AfterSellDateWithMinQuality()
        {
            IList<Item> items = new List<Item> { new Item { Name = "randomstring", DaysRemaining = -10, Quality = 0 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(-11, items[0].DaysRemaining);
            Assert.Equal(0, items[0].Quality);
        }

        [Fact]
        public void AfterSellDateNearMinQuality()
        {
            IList<Item> items = new List<Item> { new Item { Name = "randomstring", DaysRemaining = -10, Quality = 1 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(-11, items[0].DaysRemaining);
            Assert.Equal(0, items[0].Quality);
        }
    }

    public class AgedCheddarTests
    {
        [Fact]
        public void BeforeSellDate()
        {
            IList<Item> items = new List<Item> { new Item { Name = "Aged Cheddar", DaysRemaining = 5, Quality = 10 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(4, items[0].DaysRemaining);
            Assert.Equal(11, items[0].Quality);
        }

        [Fact]
        public void WithMaxQuality()
        {
            IList<Item> items = new List<Item> { new Item { Name = "Aged Cheddar", DaysRemaining = 5, Quality = 50 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(4, items[0].DaysRemaining);
            Assert.Equal(50, items[0].Quality);
        }

        [Fact]
        public void OnSellDate()
        {
            IList<Item> items = new List<Item> { new Item { Name = "Aged Cheddar", DaysRemaining = 0, Quality = 10 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(-1, items[0].DaysRemaining);
            Assert.Equal(12, items[0].Quality);
        }

        [Fact]
        public void OnSellDateWithMaxQuality()
        {
            IList<Item> items = new List<Item> { new Item { Name = "Aged Cheddar", DaysRemaining = 0, Quality = 50 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(-1, items[0].DaysRemaining);
            Assert.Equal(50, items[0].Quality);
        }

        [Fact]
        public void OnSellDateNearMaxQuality()
        {
            IList<Item> items = new List<Item> { new Item { Name = "Aged Cheddar", DaysRemaining = 0, Quality = 49 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(-1, items[0].DaysRemaining);
            Assert.Equal(50, items[0].Quality);
        }

        [Fact]
        public void AfterSellDate()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Aged Cheddar", DaysRemaining = -10, Quality = 10 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(-11, items[0].DaysRemaining);
            Assert.Equal(12, items[0].Quality);
        }

        [Fact]
        public void AfterSellDateWithMaxQuality()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Aged Cheddar", DaysRemaining = -10, Quality = 50 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(-11, items[0].DaysRemaining);
            Assert.Equal(50, items[0].Quality);
        }

        [Fact]
        public void AfterSellDateNearMaxQuality()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Aged Cheddar", DaysRemaining = -10, Quality = 49 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(-11, items[0].DaysRemaining);
            Assert.Equal(50, items[0].Quality);
        }
    }

    public class HammerTests
    {
        [Fact]
        public void hammer_BeforeSellDate()
        {
            IList<Item> items = new List<Item> { new Item { Name = "Hammer", DaysRemaining = 5, Quality = 40 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(5, items[0].DaysRemaining);
            Assert.Equal(40, items[0].Quality);
        }

        [Fact]
        public void hammer_OnSellDate()
        {
            IList<Item> items = new List<Item> { new Item { Name = "Hammer", DaysRemaining = 0, Quality = 40 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(0, items[0].DaysRemaining);
            Assert.Equal(40, items[0].Quality);
        }

        [Fact]
        public void hammer_AfterSellDate()
        {
            IList<Item> items = new List<Item> { new Item { Name = "Hammer", DaysRemaining = -10, Quality = 40 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(-10, items[0].DaysRemaining);
            Assert.Equal(40, items[0].Quality);
        }
    }

    public class ConcertTicketsTests
    {
        [Fact]
        public void LongBeforeSellDate()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = 11, Quality = 10 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(10, items[0].DaysRemaining);
            Assert.Equal(11, items[0].Quality);
        }

        [Fact]
        public void LongBeforeSellDateAtMaxQuality()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = 11, Quality = 50 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(10, items[0].DaysRemaining);
            Assert.Equal(50, items[0].Quality);
        }

        [Fact]
        public void MediumCloseToSellDateUpperBound()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = 10, Quality = 10 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(9, items[0].DaysRemaining);
            Assert.Equal(12, items[0].Quality);
        }

        [Fact]
        public void MediumCloseToSellDateUpperBoundAtMaxQuality()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = 10, Quality = 50 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(9, items[0].DaysRemaining);
            Assert.Equal(50, items[0].Quality);
        }

        [Fact]
        public void MediumCloseToSellDateUpperBoundNearMaxQuality()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = 10, Quality = 49 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(9, items[0].DaysRemaining);
            Assert.Equal(50, items[0].Quality);
        }

        [Fact]
        public void MediumCloseToSellDateLowerBound()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = 6, Quality = 10 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(5, items[0].DaysRemaining);
            Assert.Equal(12, items[0].Quality);
        }

        [Fact]
        public void MediumCloseToSellDateLowerBoundAtMaxQuality()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = 6, Quality = 50 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(5, items[0].DaysRemaining);
            Assert.Equal(50, items[0].Quality);
        }

        [Fact]
        public void MediumCloseToSellDateLowerBoundNearMaxQuality()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = 6, Quality = 49 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(5, items[0].DaysRemaining);
            Assert.Equal(50, items[0].Quality);
        }

        [Fact]
        public void VeryCloseToSellDateUpperBound()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = 5, Quality = 10 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(4, items[0].DaysRemaining);
            Assert.Equal(13, items[0].Quality);
        }

        [Fact]
        public void VeryCloseToSellDateUpperBoundAtMaxQuality()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = 5, Quality = 50 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(4, items[0].DaysRemaining);
            Assert.Equal(50, items[0].Quality);
        }

        [Fact]
        public void VeryCloseToSellDateUpperBoundNearMaxQuality()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = 5, Quality = 48 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(4, items[0].DaysRemaining);
            Assert.Equal(50, items[0].Quality);
        }

        [Fact]
        public void VeryCloseToSellDateLowerBound()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = 1, Quality = 10 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(0, items[0].DaysRemaining);
            Assert.Equal(13, items[0].Quality);
        }

        [Fact]
        public void VeryCloseToSellDateLowerBoundAtMaxQuality()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = 1, Quality = 50 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(0, items[0].DaysRemaining);
            Assert.Equal(50, items[0].Quality);
        }

        [Fact]
        public void VeryCloseToSellDateLowerBoundNearMaxQuality()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = 1, Quality = 48 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(0, items[0].DaysRemaining);
            Assert.Equal(50, items[0].Quality);
        }

        [Fact]
        public void OnSellDate()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = 0, Quality = 50 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(-1, items[0].DaysRemaining);
            Assert.Equal(0, items[0].Quality);
        }

        [Fact]
        public void AfterSellDate()
        {
            IList<Item> items = new List<Item>
                { new Item { Name = "Concert Tickets", DaysRemaining = -10, Quality = 10 } };
            new GildedRose().ProcessEndOfDay(items);

            Assert.Equal(-11, items[0].DaysRemaining);
            Assert.Equal(0, items[0].Quality);
        }
    }
}